tests/unit/policyEvaluator/requestUtils.spec.js

```diff
@@ -110,3 +111,79 @@ describe('requestUtils.getClientIp', () => {
         assert.strictEqual(result, dummyRemoteIP);
     });
 });
+
+describe('requestUtils.getHttpProtocolSecurity', () => {
+    const configWithProxy = require('../../utils/dummyS3ConfigProxy.json');
+    const configWithoutProxy = require('../../utils/dummyS3Config.json');
+    const testClientIp = '192.168.100.1';
+    const testProxyIp = '192.168.100.2';
+
+    it('should return true if request comes via trusted proxy with https proto header', () => {
+        const request = new DummyRequest({
+            headers: {
+                'x-forwarded-proto': 'https',
+            },
+            socket: {
+                remoteAddress: testProxyIp,
+            },
+        });
+        const result = requestUtils.getHttpProtocolSecurity(request, configWithProxy);
+        assert.strictEqual(result, true);
+    });
+
+    it('should return false if request comes via trusted proxy with http proto header', () => {
+        const request = new DummyRequest({
+            headers: {
+                'x-forwarded-proto': 'http',
+            },
+            socket: {
+                remoteAddress: testProxyIp,
+            },
+        });
+        const result = requestUtils.getHttpProtocolSecurity(request, configWithProxy);
+        assert.strictEqual(result, false);
+    });
+
+    it('should check TLS when request not from trusted proxy', () => {
+        const request = new DummyRequest({
+            headers: {
+                'x-forwarded-proto': 'https',
+            },
+            socket: new TLSSocket(null),
+        });
+        request.socket.encrypted = true;
+        const result = requestUtils.getHttpProtocolSecurity(request, configWithoutProxy);
+        assert.strictEqual(result, true);
+    });
+
+    it('should return false for non-TLS socket', () => {
+        const request = new DummyRequest({
+            headers: {},
```

Comment:
should probably set `x-forewarded-proto = https` here as well, to explicitely test that we ignore it in that case.